### PR TITLE
 Add Archiver that creates modular JAR files using the JDK jar tool

### PR DESCRIPTION
**Pull request by plamentotev**
As discussed on dev list I've created `Archiver` that allows the creation of modular JAR files and setting their module version and main class. The implementation uses `java.util.spi.ToolProvider` so Java 9+ is required (it will run on Java 7+ but the version and main class will not be set). As discussed it does not support tool chain, but it is easy to add support if you think it is required.

This PR depends on #83. As new method is added to the API i decided it's better to split it into two merge requests. If they approved I'll rebase and merge into master

/cc @rfscholte

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
   </parent>
 
   <artifactId>plexus-archiver</artifactId>
-  <version>3.5.1-SNAPSHOT</version>
+  <version>3.6-SNAPSHOT</version>
```
Comment (khmarbaise) [inline pom.xml]: Can we do 3.6.0-SNAPSHOT instead ? I would suggest to have three digits.. ? WDYT ?

Comment (plamentotev) [reply pom.xml]: The last couple of releases are two digits. I don't know if there is any particular reason - I think @michael-o did the last release. Maybe he knows. Other than that I agree that with three digits is better.

Comment (michael-o) [reply pom.xml]: We have agreed  at some point to use semver with three digits.

**Comment by michael-o** [comment]
Can you please squash before merging?

File: pom.xml (hunk 1 above)
Comment (plamentotev) [reply pom.xml]: ok, then I'll change it to 3.6.0

**Comment by plamentotev** [comment]
@michael-o I'll squash it. You mean that 38e2880, 54fe62b and 3c2dc6a should be one commit or that both PRs should be one commit?

**Comment by khmarbaise** [comment]
Yes that means squashing ...to have only a single commit which represents the change and after merging it to master it's easier to understand in the history...

**Pull request closed without merging**